### PR TITLE
update docstring of algorithms

### LINKED ISSUE #143: add docstring suggestion to discretize
See [here](https://github.com/JuliaReach/MathematicalSystems.jl/pull/122) for the suggestions.

### PR DESCRIPTION
**Pull request by ueliwechsler**
close #143

### CODE REVIEW

File: src/discretize.jl [hunk 1]
```diff
@@ -81,14 +81,21 @@ Exact discretization algorithm for affine systems.
 
 ### Algorithm
 
-Consider a `NoisyAffineControlledContinuousSystem` with system dynamics
-``x' = Ax + Bu + c + Dw``.
-
-The exact discretization is calculated by solving the integral for
-``t = [t, t + ΔT]`` for a fixed input `u` and fixed noise realization `w`.
+This algorithm consists of integrating the continuous differential equation over
+a specified time interval to obtain an associated discrete differential equation.
```
Comment (schillic) [inline src/discretize.jl]:
```suggestion
a specified time interval to obtain an associated recurrence relation.
```
While the old term exists (and also the term "difference relation"), I prefer [recurrence relation](https://en.wikipedia.org/wiki/Recurrence_relation).

File: src/discretize.jl [hunk 2]
```diff
@@ -106,14 +113,20 @@ Euler discretization algorithm for affine systems.
 
 ### Algorithm
 
-Consider a `NoisyAffineControlledContinuousSystem` with system dynamics
-``x' = Ax + Bu + c + Dw``.
+This algorithm consists of a first order approximation to obtain an associated
```
Comment (schillic) [inline src/discretize.jl]:
```suggestion
This algorithm consists of a first-order approximation to obtain an associated
```

File: src/discretize.jl [hunk 3]
```diff
@@ -106,14 +113,20 @@ Euler discretization algorithm for affine systems.
...
+discrete differential equation. The algorithm applies to any system of the form
```
Comment (schillic) [inline src/discretize.jl]:
```suggestion
recurrence relation. The algorithm applies to any system of the form
```

File: src/discretize.jl [hunk 4]
```diff
@@ -81,14 +81,21 @@ Exact discretization algorithm for affine systems.
...
 
-Consider a `NoisyAffineControlledContinuousSystem` with system dynamics
-``x' = Ax + Bu + c + Dw``.
-
-The exact discretization is calculated by solving the integral for
-``t = [t, t + ΔT]`` for a fixed input `u` and fixed noise realization `w`.
+This algorithm consists of integrating the continuous differential equation over
+a specified time interval to obtain an associated discrete differential equation.
+The algorithm applies to any system of the form `x' = Ax + Bu + c + Dw` where the
+state matrix `A` is invertible, and other system types e.g. linear systems,
```
Comment (schillic) [inline src/discretize.jl]:
```suggestion
state matrix `A` is invertible, and other system types, e.g. linear systems
```

File: src/discretize.jl [hunk 5]
```diff
@@ -106,14 +113,20 @@ Euler discretization algorithm for affine systems.
 
 ### Algorithm
 
-Consider a `NoisyAffineControlledContinuousSystem` with system dynamics
-``x' = Ax + Bu + c + Dw``.
+This algorithm consists of a first order approximation to obtain an associated
+discrete differential equation. The algorithm applies to any system of the form
+`x' = Ax + Bu + c + Dw`, and other system types e.g. linear systems, `x' = Ax`
```
Comment (schillic) [inline src/discretize.jl]:
```suggestion
`x' = Ax + Bu + c + Dw`, and other system types, e.g. linear systems `x' = Ax`
```

File: src/discretize.jl [hunk 6]
```diff
@@ -81,14 +81,21 @@ Exact discretization algorithm for affine systems.
...
-
-The exact discretization is calculated by solving the integral for
-``t = [t, t + ΔT]`` for a fixed input `u` and fixed noise realization `w`.
+This algorithm consists of integrating the continuous differential equation over
+a specified time interval to obtain an associated discrete differential equation.
+The algorithm applies to any system of the form `x' = Ax + Bu + c + Dw` where the
+state matrix `A` is invertible, and other system types e.g. linear systems,
+`x' = Ax` which are included in the above formulation.
+
+Without loss of generality, consider a  `NoisyAffineControlledContinuousSystem`
```
Comment (schillic) [inline src/discretize.jl]:
```suggestion
Without loss of generality, consider a `NoisyAffineControlledContinuousSystem`
```

File: src/discretize.jl [hunk 7]
```diff
@@ -106,14 +113,20 @@ Euler discretization algorithm for affine systems.
...
 ### Algorithm
 
-Consider a `NoisyAffineControlledContinuousSystem` with system dynamics
-``x' = Ax + Bu + c + Dw``.
+This algorithm consists of a first order approximation to obtain an associated
+discrete differential equation. The algorithm applies to any system of the form
+`x' = Ax + Bu + c + Dw`, and other system types e.g. linear systems, `x' = Ax`
+which are included in the above formulation.
+
+Without loss of generality, consider a  `NoisyAffineControlledContinuousSystem`
```
Comment (schillic) [inline src/discretize.jl]:
```suggestion
Without loss of generality, consider a `NoisyAffineControlledContinuousSystem`
```

File: src/discretize.jl [hunk 8]
```diff
@@ -106,14 +113,20 @@ Euler discretization algorithm for affine systems.
...
+with system dynamics ``x' = Ax + Bu + c + Dw``.
 
-The Euler discretization, a first order approximation of the exact
-discretization [`ExactDiscretization`](@ref), writes as
+The Euler discretization is calculated by taking the first order approximation
```
Comment (schillic) [inline src/discretize.jl]:
```suggestion
The Euler discretization is calculated by taking the first-order approximation
```

File: src/discretize.jl [hunk 9]
```diff
@@ -106,14 +113,20 @@ Euler discretization algorithm for affine systems.
...
+of the exact discretization [`ExactDiscretization`](@ref).
+The resulting discretization writes as
 ``x^+ = A^d x + B^d u + c^d + D^d w``
-where  ``A^d = I + ΔT \\cdot A``, ``B^d = ΔT \\cdot B``,
-``c^d = ΔT \\cdot c`` and ``D^d = ΔT \\cdot D``.
+where  ``A^d = I + ΔT ~  A``, ``B^d = ΔT ~  B``,
```
Comment (schillic) [inline src/discretize.jl]:
```suggestion
where  ``A^d = I + ΔT ~ A``, ``B^d = ΔT ~ B``,
```

File: src/discretize.jl [hunk 10]
```diff
@@ -106,14 +113,20 @@ Euler discretization algorithm for affine systems.
...
+``c^d = ΔT ~  c`` and ``D^d = ΔT ~  D``.
```
Comment (schillic) [inline src/discretize.jl]:
```suggestion
``c^d = ΔT ~ c`` and ``D^d = ΔT ~ D``.
```

File: src/discretize.jl (hunk 1 above)
Comment (ueliwechsler) [reply src/discretize.jl]: I have never heard `recurrence relation` in this context before.  But yeah, also makes sense. I am use to the term `difference equation` . Could this be an alternative?

Comment (schillic) [reply src/discretize.jl]: Yeah, I meant "difference *equation*". In the end, I am fine with everything, it was just a suggestion. I would get confused if I read "discrete differential". Are there other opinions?

Comment (ueliwechsler) [reply src/discretize.jl]: I can't think of any at the moment.

File: src/discretize.jl [hunk 11]
```diff
@@ -82,7 +82,7 @@ Exact discretization algorithm for affine systems.
 ### Algorithm
 
 This algorithm consists of integrating the continuous differential equation over
-a specified time interval to obtain an associated discrete differential equation.
+a specified time interval to obtain an difference differential equation.
```
Comment (schillic) [inline src/discretize.jl]:
```suggestion
a specified time interval to obtain an associated difference equation.
```

File: src/discretize.jl [hunk 12]
```diff
@@ -114,7 +114,7 @@ Euler discretization algorithm for affine systems.
 ### Algorithm
 
 This algorithm consists of a first-order approximation to obtain an associated
-discrete differential equation. The algorithm applies to any system of the form
+ difference equation. The algorithm applies to any system of the form
```
Comment (schillic) [inline src/discretize.jl]:
```suggestion
difference equation. The algorithm applies to any system of the form
```

File: src/discretize.jl (hunk 1 above)
Comment (schillic) [reply src/discretize.jl]: *opinions*, not *options* :wink:
@blegat @mforets?

Comment (mforets) [reply src/discretize.jl]: In the context of this library, we call such systems discrete systems. I would use that name. Bu both the terms "Discrete differential equations" and "recurrence relation" are also correct imo.

**Pull request merged**